nmma/pbilby/parser/analysis.py

```diff
@@ -12,7 +12,7 @@ def create_nmma_analysis_parser(sampler="dynesty"):
 
     analysis_parser = argparse.ArgumentParser(prog="nmma_analysis", parents=[parser])
     analysis_parser.add_argument(
-        "data-dump",
```

Comment:
This just means --data-dump is the argument and then the argument is args.data_dump already.

Reply:
but the data-dump is a positional argument and therefore it gets passed as args.data-dump which causes trouble

Reply:
No I think it turns into data_dump. Can @sahiljhawar confirm the behavior perhaps?

Reply:
well i just checked again and when i change it back to data-dump it becomes args.data-dump at least in pbilby/analysis/main.
![Screenshot from 2023-09-28 20-14-35](https://github.com/nuclear-multimessenger-astronomy/nmma/assets/136069487/6e93f84d-8e7f-4ef2-ae0d-a3f46219a0c5)

Reply:
No, it does not gets changed to `data_dump`. It remains data-dump and it cannot be accessed as an attribute of `args`

Reply:
I would suggest to change data_dump from position argument (`data_dump`) to a normal argument (`--data-dump`) and see if that works.

Reply:
@haukekoehn @sahiljhawar Ah that is presumably the difference. Sorry, I didn't realize that it maintains its name if positional.

Reply:
@haukekoehn `args.data-dump` is not a valid syntax in case the value of `data-dump` needs to be extracted it should be done as such `getattr(args, "data-dump"))`. Also, if positional arguments are being used then in CLI, then only the order matters and neither of the two works: `--data-dump`, `--data_dump`

Reply:
yeah now i changed it to --data-dump